File: app/services/bing_service.rb
class BingService
  def get_news(title)
    @resp = Faraday.get 'https://api.datamarket.azure.com/Bing/Search/v1/Composite' do |req|
      req.params["Sources"] = "'news'"
      req.params["Query"] = "'" + "#{title}" + "'"
      req.params['$format'] = 'json'
      req.headers['Authorization'] = 'Basic ' + "#{ENV['BING_BASE64']}"
    end
    @result = JSON.parse(@resp.body)
  end
end

File: app/assets/javascripts/app/services/BingService.js
function BingService($http){
  
  this.getNews = function(title){
    return $http.get('http://gamefeedapp.herokuapp.com/api/v1/bing/news.json', 
      {params:
        {title: title}
      }
    );
  }
}

BingService.$inject = ['$http'];

angular
  .module('app')
  .service('BingService', BingService)
